impress-org/givewp
# Feature: add Option-Based Form Editor feature flag

Resolves [GIVE-1264]

## Description



This PR adds a new option to the `Give > Settings > Advanced` page that allows users to enable or disable the Option-Based Form Editor. This new option is enabled by default if the website has any Optio-Based forms on DB and disabled by default if it has only Visual Form Builder forms on DB.

## Affects



The "Settings" page and the "Add campaign form" button

## Visuals



![image](https://github.com/user-attachments/assets/3918c165-e070-4d12-92a4-b01e3136c6bb)

![image](https://github.com/user-attachments/assets/3b26319c-9d87-4d2d-a407-f4b9a92d8301)

![image](https://github.com/user-attachments/assets/a9970b36-935b-48cd-b03c-afad36502638)

![image](https://github.com/user-attachments/assets/7bb61907-7a63-4a95-b2d4-7c9c86a815f9)


## Testing Instructions



Go to the `Give > Settings > Advanced` page to enable or disable the `Option-Based Form Editor` option and make sure the expected behaviors occur.

**Expected behaviors with the legacy editor disabled:** 

- Options used only by the legacy editor are hidden.
- When the user clicks on the "Add campaign form" button, the Visual Form Builder is used by default.

**Expected behaviors with the legacy editor enabled:** 

- Options used only by the legacy editor have a helper text net to the label.
- When the user clicks on the "Add campaign form" button, a modal that allows selecting which editor should be used, is displayed.

**Expected behaviors in any scenario:** 

In both editors, you can find a link to return to the campaign overview page.

## Pre-review Checklist






[GIVE-1264]: https://stellarwp.atlassian.net/browse/GIVE-1264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
